src/test/java/net/minecraftforge/debug/item/ItemPreventPickupTest.java

```diff
@@ -0,0 +1,125 @@
+    public static final RegistryObject<Item> DUMMY = ITEMS.register("dummy", () -> new Item(new Item.Properties()) {
+        @Override
+        public void appendHoverText(ItemStack stack, @Nullable Level level, List<Component> list, TooltipFlag flag)
+        {
+            super.appendHoverText(stack, level, list, flag);
+
+            if(ENABLE)
+            {
+                list.add(Component.literal("Valid pickup states:"));
+                list.add(Component.literal("Player: Only If Sneaking"));
+                list.add(Component.literal("Zombie: If standing on stone"));
+                list.add(Component.literal("Hopper: Only if stacked more than 1"));
+            }
+        }
+    });
+
+    public static final ItemPickupQuery IF_MORE_THAN_ONE = (item, stack, level, pos, collector, pickupReasons) -> stack.getCount() >= 2;
+
+    public ItemPreventPickupTest()
+    {
+        var modBus = FMLJavaModLoadingContext.get().getModEventBus();
+        ITEMS.register(modBus);
+
+        // event used to add dummy item to MISC / UTILITIES tab
+        modBus.addListener(this::onBuildCreativeModeTabContents);
+        // event used to add tooltips, used to display how items are allowed to be picked up
+        MinecraftForge.EVENT_BUS.addListener(this::onItemTooltip);
+        // event used to disable hopper pickup type for diamond items
+        MinecraftForge.EVENT_BUS.addListener(this::onItemTossed);
+        // event used to add item pickup query for dummy item (hopper pickup only)
+        // high priority to ensure we add queries before they are tested against
+        MinecraftForge.EVENT_BUS.addListener(EventPriority.HIGH, this::onItemPickupAllowed_Pre);
+        // event used to forcibly allow/disallow items from being picked up, based on a few states & item types
+        MinecraftForge.EVENT_BUS.addListener(this::onItemPickupAllowed);
+
+    }
+
+    private void onBuildCreativeModeTabContents(CreativeModeTabEvent.BuildContents event)
+    {
+        if(event.getTab() == CreativeModeTabs.TOOLS_AND_UTILITIES) event.accept(DUMMY);
+    }
+
+    private void onItemPickupAllowed_Pre(ItemPickupAllowedEvent event)
+    {
+        // only for new dummy item, and only add the queries once per item entity
+        if(ENABLE && event.getItemStack().is(DUMMY.get()) && !event.getEntity().getPersistentData().getBoolean("ForgeInitItemPickupQuery"))
+        {
+            // hoppers only pickup if item is stacked more than 1
+            if(event.hasPickupReason(ItemPickupReasons.HOPPER)) event.getEntity().addPickupQuery(IF_MORE_THAN_ONE);
+            event.getEntity().getPersistentData().putBoolean("ForgeInitItemPickupQuery", true);
+        }
+    }
+
+    private void onItemPickupAllowed(ItemPickupAllowedEvent event)
+    {
+        if(ENABLE)
+        {
+            // forcibly disallow picking up iron ingots
+            if(event.getItemStack().is(Tags.Items.INGOTS_IRON)) event.setResult(Event.Result.DENY);
```

Comment:
Formatting: for `if` statements with `else` clauses, please move all code statements to new lines instead of being inline with the `if`/`else`.